src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.DebugVerifier.cs

```diff
@@ -186,6 +186,18 @@ private void VerifyExpression(BoundExpression expression, bool overrideSkippedEx
                 return base.VisitAssignmentOperator(node);
             }
 
+            public override BoundNode? VisitCompoundAssignmentOperator(BoundCompoundAssignmentOperator node)
+            {
+                if (node.LeftConversion is BoundConversion leftConversion)
```

Comment:
`BoundCompoundAssignmentOperator` also has a `FinalConversion` (for converting result of binary operator back to left type). Should that be verified here too?

Reply:
The fact that we are verifying this here is somewhat an accident of how the shared `ReinferAndVisitBinaryOperator` works. We didn't end up needing to do that for the final conversion. The way we happen to visit the final conversion doesn't end up including it in the visited nodes map.

Reply:
> [...] doesn't end up including [the final conversion] in the visited nodes map.

That seems strange. I would have expected the visited nodes map to include all bound nodes, with enumerated exceptions (in `s_skippedExpressions`).

Reply:
Some nodes, including the Conversion/Placeholder nodes, are marked "skipInVisitor" and won't be visited unless we go out of our way to do so.